sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -208,43 +246,13 @@ class SQLAppStatusListener(
       stageId: Int,
       attemptId: Int,
       taskId: Long,
+      partIdx: Int,
```

Comment:
`partIdx` -> `taskIdx`.

Reply:
No. This is the index of the *partition* being computed.

Reply:
So this is what happens.

When you don't have speculative tasks, then that comment in `TaskInfo` is true; in that it's the task's index in the task set.

When you add speculative tasks, that becomes false; the new task will have the same index as some other task in the same task set. Speculation might have been added after `TaskInfo` and the comment never updated.

So the gist is that this `index` is basically a proxy for the partition index. Two tasks with the same index are computing the same partition. So I think it's more important to track the intent (that this field refers, even if indirectly, to what partition is being calculated) than trying to be exact about its meaning.